cupy/sorting/search.py

```diff
@@ -187,55 +187,78 @@ def where(condition, x=None, y=None):
 _preamble = '''
 template<typename T>
 __device__ bool _isnan(T val) {
-    return false;
-}
-template<>
-__device__ bool _isnan(float16 val) {
-    return isnan(val);
-}
-template<>
-__device__ bool _isnan(float val) {
-    return isnan(val);
-}
-template<>
-__device__ bool _isnan(double val) {
-    return isnan(val);
-}
-template<>
-__device__ bool _isnan(const complex<double>& val) {
-    return isnan(val);
-}
-template<>
-__device__ bool _isnan(const complex<float>& val) {
-    return isnan(val);
+    return val != val;
 }
 '''
 
 
 _searchsorted_kernel = core.ElementwiseKernel(
-    'S x, raw T bins, int64 n_bins, bool side_is_right',
+    'S x, raw T bins, int64 n_bins, bool side_is_right, bool check_mono',
```

Comment:
How about this (note that logic is inverted)? "check" is ambiguous.
```suggestion
    'S x, raw T bins, int64 n_bins, bool side_is_right, bool assume_increasing',
```
Write a comment to describe this argument.